Sources/SwiftDocC/SourceRepository/SourceRepository.swift

```diff
@@ -89,4 +89,16 @@ public extension SourceRepository {
             formatLineNumber: { line in "lines-\(line)" }
         )
     }
+    
+    /// Creates a source repository hosted by the device's filesystem.
+    ///
+    /// Use this source repository to format `file://` links to files on the
+    /// device where documentation is being presented.
+    static func localFilesystem() -> SourceRepository {
+        SourceRepository(
+            checkoutPath: "",
+            sourceServiceBaseURL: URL(string: "file:///")!,
+            formatLineNumber: { line in "L\(line)" }
```

Comment:
I'm not entirely opposed to this but it does seem pretty specific to the GitHub/Stash usecase and less relevant for the file:// use case.

Maybe a more verbose but readable query parameter would be appropriate here?

```
file:///my/file/path/file.swift?line=39
```

Reply:
Using fragments to indicate a position in a document is a well-established pattern, but since there's no real convention for filesystem documents, I'm fine with either approach. Maybe we could remove the `L`. Using a query parameter would need additional support from `SourceRepository`, the `formatLineNumber` closure writes a fragment.

Reply:
> Using fragments to indicate a position in a document is a well-established pattern

That's good to know. I was looking for standardized formats for file URLs that include positioning and couldn't find any so I'd be curious to see some examples.

I like the idea of removing the L – it just adds additional work on the client side to parse it for no real benefit from what I can see.

Reply:
I haven't seen column information getting encoded in the URL. but that's a good point, should we keep the `L` so that we can add `C` later on if we'd like to? `#L23C53`